types/react/global.d.ts

```diff
@@ -0,0 +1,176 @@
+/*
+React projects that don't include the DOM library need these interfaces to compile.
+React Native applications use React, but there is no DOM available. The JavaScript runtime
```

Comment:
> React projects that don't include the DOM library need these interfaces to compile.

How common is this?

> React Native applications use React, but there is no DOM available. The JavaScript runtime

should this go into React Native .d.ts instead then?

Reply:
> How common is this?

Currently it's not possible, but it is a goal to allow React Native projects to compile without the DOM since it would allow code to compile that cannot run. I don't know how many other environments there are like this, but I suspect that React code running in Node exclusively for server-side rendering may also want to leave the DOM APIs out.
To quote from the original [React 14 announcement](https://facebook.github.io/react/blog/2015/07/03/react-v0.14-beta-1.html):
_As we look at packages like react-native, react-art, react-canvas, and react-three, it's become clear that the beauty and essence of React has nothing to do with browsers or the DOM._

I haven't used those other projects myself so I'm not sure how relevant they are or even if they use/support TypeScript.

>should this go into React Native .d.ts instead then?

It could, but since React does support running in environments outside the browser, the case could be made that the React types should also support compiling without `--lib dom`. It would also allow any future DOM dependencies added to React's `index.d.ts` to also include the placeholder declaration in `global.d.ts` so that non-DOM users of the type won't become broken when upgrading.